src/app/admin/product-form/product-form.component.ts

```diff
@@ -38,4 +38,10 @@ export class ProductFormComponent {
 
     this.router.navigate(['/admin/products']);
   }
+
+  delete() {
+    if (!confirm('Are you sure you want to delete this product?')) return;
```

Comment:
can you pull this error message into a const at the top of the file

Reply:
Done.